gadget/install/partition.go

```diff
@@ -149,20 +149,26 @@ func buildPartitionList(dl *gadget.OnDiskVolume, vol *gadget.Volume, opts *Creat
 	// called before this function. muinstaller also checks for
 	// PartialStructure before this is run.
 	pIndex := 0
-
-	// Keep track what partitions we already have on disk - the keys to this map
-	// is the starting sector of the structure we have seen.
-	// TODO: use quantity.SectorOffset or similar when that is available
-
-	seen := map[uint64]bool{}
 	for _, s := range dl.Structure {
-		start := uint64(s.StartOffset) / sectorSize
-		seen[start] = true
 		if s.DiskIndex > pIndex {
 			pIndex = s.DiskIndex
 		}
 	}
 
+	// Find out partitions already on the disk, if we don't want to create
+	// all. If CreateAllMissingPartitions is set we are being called from
+	// muinstaller and no partitions are expected on the disk.
+	// TODO we should avoid using createMissingPartitions as ancillary
+	// method from muinstaller to avoid this sort of situation.
+	matchedStructs := map[int]*gadget.OnDiskStructure{}
+	if !opts.CreateAllMissingPartitions {
+		if matchedStructs, err = gadget.EnsureVolumeCompatibility(vol, dl, nil); err != nil {
```

Comment:
remind me, what happens with EnsureVolumeCompatibility when there are missing partitions? does the test you added considers that? anyway it's probably worth having a comment here about the behavior with those for future readers

Reply:
It won't error if the `AssumeCreatablePartitionsCreated` option is false, as is the case. The test considers that as save/data partitions are not created in the mocked disk (`scriptPartitionsBiosSeed`), and no error is expected.